api/taskcluster/webhook.go

```diff
@@ -194,12 +204,40 @@ func shouldProcessStatus(log shared.Logger, processAllBranches bool, status *sta
 	return true
 }
 
-func extractTaskGroupID(targetURL string) string {
-	lastSlash := strings.LastIndex(targetURL, "/")
-	if lastSlash == -1 {
-		return ""
+// handleCheckRunEvent processes an Azure Pipelines check run "completed" event.
+func handleCheckRunEvent(aeAPI shared.AppEngineAPI, event *github.CheckRunEvent) (bool, error) {
+	log := shared.GetLogger(aeAPI.Context())
+	status := event.GetCheckRun().GetStatus()
+	if status != "completed" {
+		log.Infof("Ignoring non-completed status %s", status)
+		return false, nil
 	}
-	return targetURL[lastSlash+1:]
+	detailsURL := event.GetCheckRun().GetDetailsURL()
+	sha := event.GetCheckRun().GetHeadSHA()
+
+	labels := mapset.NewSet()
+	sender := event.GetSender().GetLogin()
+	if sender != "" {
+		labels.Add(shared.GetUserLabel(sender))
+	}
+
+	taskGroupID, taskID := extractTaskGroupID(detailsURL)
+	if taskGroupID == "" {
+		return false, fmt.Errorf("unrecognized target_url: %s", detailsURL)
+	}
+
+	return processTaskclusterBuild(aeAPI.Context(), taskGroupID, taskID, sha, shared.ToStringSlice(labels)...)
+}
+
+func extractTaskGroupID(targetURL string) (string, string) {
+	regex := regexp.MustCompile("(?:/task-group-inspector/#/|/groups/)([^/]*)(?:/tasks/([^/]*))?")
```

Comment:
I'd prefer two simpler, stricter regexes instead of one with two clauses with two options each (i.e. four total combinations).

Reply:
Done.